### PR TITLE
Reduce allocations by using newtypes when possible, solves #8

### PR DESCRIPTION
**Pull request by dredozubov**

### CODE REVIEW

File: src/Data/THGen/XML.hs [hunk 1]
```diff
@@ -311,11 +313,12 @@ isoXmlGenerateEnum (ExhaustivenessName strName' exh) enumCons = do
 isoXmlGenerateRecord :: PrefixName -> [IsoXmlDescRecordPart] -> TH.DecsQ
 isoXmlGenerateRecord (PrefixName strName' strPrefix') descRecordParts = do
   let
+    isNewtype     = length descRecordParts == 1
```
Comment (4e6) [inline src/Data/THGen/XML.hs]: Something like `hasSingleField` would be more clear

File: src/Data/THGen/XML.hs [hunk 2]
```diff
@@ -311,11 +313,12 @@ isoXmlGenerateEnum (ExhaustivenessName strName' exh) enumCons = do
 isoXmlGenerateRecord :: PrefixName -> [IsoXmlDescRecordPart] -> TH.DecsQ
```
Comment (4e6) [inline src/Data/THGen/XML.hs]: It is no longer a record

File: src/Data/THGen/XML.hs [hunk 3]
```diff
@@ -341,11 +345,11 @@ isoXmlGenerateRecord (PrefixName strName' strPrefix') descRecordParts = do
...
             in
               varStrictType fName (strictType fType)
-    dataD
-      name
-      [TH.recC name fields]
-      [''Eq, ''Show]
-  lensDecls <- makeFieldOpticsForDec lensRules dataDecl
+    if isNewtype
+    -- generate a newtype instead to do less allocations later
+    then newtypeD name (TH.recC name fields) [''Eq, ''Show]
```
Comment (4e6) [inline src/Data/THGen/XML.hs]: Ok, if it remains compatible with existing lenses.

File: src/Data/THGen/XML.hs (hunk 2 above)
Comment (dredozubov) [reply src/Data/THGen/XML.hs]: fair enough

File: src/Data/THGen/XML.hs (hunk 1 above)
Comment (dredozubov) [reply src/Data/THGen/XML.hs]: This check was used at least twice: once to clarify how to construct a type definition, an another to supply fields in a different manner. The second usage differs for newtype/data, but have no meaningful difference for 1/more fields. This is why I find the name more appropriate.

**Pull request merged**